core/src/main/java/org/apache/accumulo/core/client/admin/NewTableConfiguration.java

```diff
@@ -168,7 +168,7 @@ public Map<String,String> getProperties() {
     Map<String,String> propertyMap = new HashMap<>();
 
     if (limitVersion) {
-      propertyMap.putAll(IteratorConfigUtil.generateInitialTableProperties(limitVersion));
+      propertyMap.putAll(IteratorConfigUtil.getInitialTableProperties());
     }
```

Comment:
Currently also adds DefaultKeySizeConstraint, so if `withoutDefaultIterators()` is specified, DefaultKeySizeConstraint won't be added to config. Is this intended?

Reply:
If we deprecate API methods, may want to deprecate the shells `-ndi` option and replace with a `--no-default-table-props` option or something like that, to get the behavior and the option names to align.

Reply:
I deprecated `withoutDefaultIterators()` and `-ndi` option and added `withoutDefaults()` and `-ndtp` in 797e39102700e1d187635fdd7d0fe0c0a8d1c721 and 18914a58d1206bed87cc2d6e8e299d629607a3c2 (same functionality as before, but more appropriate name)

Also considered deprecating `withoutDefaultIterators()` and:
* adding both `withoutDefaults()` and `withoutDefaultIters()` but thought that would probably add too much bloat (probably fine if there's desire for it)
* adding only `withoutDefaultIters()`, but like you said we probably don't want to change functionality